v2/pkg/engine/datasource/grpc_datasource/execution_plan.go

```diff
@@ -727,20 +796,92 @@ func (r *rpcPlanningContext) resolveServiceName(subgraphName string) string {
 	return r.mapping.Service
 }
 
-type resolvedField struct {
+type resolverField struct {
 	callerRef              int
-	parentTypeRef          int
+	parentTypeNode         ast.Node
 	fieldRef               int
 	fieldDefinitionTypeRef int
 	fieldsSelectionSetRef  int
 	responsePath           ast.Path
 
-	contextFields  []contextField
-	fieldArguments []fieldArgument
+	contextFields      []contextField
+	fieldArguments     []fieldArgument
+	fragmentSelections []fragmentSelection
+	fragmentType       OneOfType
+	memberTypes        []string
+}
+
+type fragmentSelection struct {
+	typeName        string
+	selectionSetRef int
+}
+
+// enterResolverCompositeSelectionSet handles logic when entering a composite selection set for a given field resolver.
+// It appends the inline fragment selections to the resolved field and sets the fragment type.
+func (r *rpcPlanningContext) enterResolverCompositeSelectionSet(oneOfType OneOfType, selectionSetRef int, resolvedField *resolverField) {
+	inlineFragSelections := r.operation.SelectionSetInlineFragmentSelections(selectionSetRef)
+	if len(inlineFragSelections) == 0 {
+		return
+	}
+
+	for _, inlineFragSelectionRef := range inlineFragSelections {
+		inlineFragRef := r.operation.Selections[inlineFragSelectionRef].Ref
+		inlinFragSelectionSetRef, ok := r.operation.InlineFragmentSelectionSet(inlineFragRef)
+		if !ok {
+			continue
+		}
+
+		resolvedField.fragmentSelections = append(resolvedField.fragmentSelections, fragmentSelection{
+			typeName:        r.operation.InlineFragmentTypeConditionNameString(inlineFragRef),
+			selectionSetRef: inlinFragSelectionSetRef,
+		})
+	}
+
+	resolvedField.fragmentType = oneOfType
+}
+
+// isFieldResolver checks if a field is a field resolver.
+func (r *rpcPlanningContext) isFieldResolver(fieldRef int, isRootField bool) bool {
+	if isRootField {
+		return false
+	}
+
+	return len(r.operation.FieldArguments(fieldRef)) > 0
+}
+
+// getCompositeType returns the most recent composite type from the type definitions.
```

Comment:
I think it should be named `isCompositeType`. Not sure about "the most recent composite type from the type definitions". It is somehow sounds weird considering the body the function that just checks `.Kind` field.

PS: I cannot do multiline comments in this file. GH has bug.

Reply:
Good point. I think the function looked differently when I first implemented it and I never updated the comment.